app/src/main/java/ch/epfl/sweng/SDP/game/drawing/DrawingOffline.java

```diff
@@ -20,72 +20,7 @@
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.ValueEventListener;
 
-public class DrawingOffline extends DrawingActivity implements SensorEventListener {
-
-    private int speed;
-    private SensorManager sensorManager;
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-
-        speed = 5; //will be passed as variable in future, not hardcoded
-
-        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
-    }
-
-    // MARK: ACCELEROMETER METHODS
-
-    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
-    @Override
-    protected void onResume() {
-        super.onResume();
-
-        // Register accelerometer sensor listener
-        Sensor accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
-        if (accelerometer != null) {
-            sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_FASTEST);
-        } else {
-            Log.d(TAG, "We couldn't find the accelerometer on device.");
-        }
-    }
-
-    /**
-     * Fires when a sensor detected a change.
-     *
-     * @param sensorEvent the sensor that has changed
-     */
-    @Override
-    public void onSensorChanged(SensorEvent sensorEvent) {
-        Sensor mySensor = sensorEvent.sensor;
-
-        // we only use accelerometer
-        if (mySensor.getType() == Sensor.TYPE_ACCELEROMETER) {
-            updateValues(sensorEvent.values[0], sensorEvent.values[1]);
-            handler.sendEmptyMessage(0);
-        }
-    }
-
-    @Override
-    public void onAccuracyChanged(Sensor sensor, int accuracy) {
-        //does nothing
-    }
-
-    /**
-     * Called when accelerometer changed, circle coordinates are updated.
-     *
-     * @param coordinateX coordiate
-     * @param coordinateY coordinate
-     */
-    public void updateValues(float coordinateX, float coordinateY) {
-        float tempX = paintView.getCircleX();
-        float tempY = paintView.getCircleY();
-
-        tempX -= coordinateX * speed;
-        tempY += coordinateY * speed;
-
-        paintView.setCircle((int) tempX, (int) tempY);
-    }
+public class DrawingOffline extends GyroDrawingActivity implements SensorEventListener {
```

Comment:
I don't think you need to add implements SensorEventListener here since GyroDrawingActivity already implements it.

Reply:
Very true